### PR TITLE
Feature/cbi 824 multi network health

### PR DESCRIPTION
**Pull request by karthikeyan-cere**
Added multi-network support for health check

### CODE REVIEW

File: src/modules/health/health.controller.ts [hunk 1]
```diff
@@ -23,60 +23,74 @@ export class HealthController {
...
   @ApiNotFoundResponse({description: 'Finalization is unhealthy.'})
-  public async finalization(@Res() res: Response): Promise<any> {
-    const diff = await this.healthService.finalization();
-    if (diff) {
-      res.status(HttpStatus.NOT_FOUND).send();
+  public async finalization(@Param('network') network: string, @Res() res: Response): Promise<any> {
+    if (!this.healthService.network.has(network)) {
+      throw new BadRequestException(`Invalid network type.`);
+    } else {
+      const diff = await this.healthService.finalization(network);
```
Comment (evgeny-s) [inline src/modules/health/health.controller.ts:50]: No need `else` here.

File: src/modules/health/health.controller.ts [hunk 2]
```diff
@@ -23,60 +23,74 @@ export class HealthController {
...
-      res.status(HttpStatus.NO_CONTENT).send();
+  public async blockProduction(@Param('network') network: string, @Res() res: Response): Promise<any> {
+    if (this.healthService.network.has(network)) {
+      const diff = await this.healthService.blockProduction(network);
+      if (diff) {
+        res.status(HttpStatus.NO_CONTENT).send();
+      }
+      res.status(HttpStatus.NOT_FOUND).send();
+    } else {
+      throw new BadRequestException(`Invalid network type.`);
```
Comment (evgeny-s) [inline src/modules/health/health.controller.ts:69]: Let's use the same approach: if not has network - throw error. And remove `else`

File: src/modules/health/health.controller.ts [hunk 3]
```diff
@@ -23,60 +23,74 @@ export class HealthController {
...
-  @Get('node-dropped')
+  @Get('node-dropped/:network')
   @ApiResponse({status: 204, description: 'Validator node is healthy.'})
   @ApiNotFoundResponse({description: 'Validator node is unhealthy.'})
-  public async nodeDropped(@Res() res: Response): Promise<any> {
-    const result = await this.healthService.nodeDropped();
-    if (result) {
-      res.status(HttpStatus.NO_CONTENT).send();
+  public async nodeDropped(@Param('network') network: string, @Res() res: Response): Promise<any> {
+    if (this.healthService.network.has(network)) {
```
Comment (evgeny-s) [inline src/modules/health/health.controller.ts:77]: The same as above

**Review by evgeny-s — CHANGES_REQUESTED** [review]
Please check comments and remove all unneeded ENV parameter, which mentioned in the [ticket](https://cerenetwork.atlassian.net/browse/CBI-824).

File: src/modules/health/health.controller.ts (hunk 1 above)
Comment (karthikeyan-cere) [reply src/modules/health/health.controller.ts]: Done

File: src/modules/health/health.controller.ts (hunk 2 above)
Comment (karthikeyan-cere) [reply src/modules/health/health.controller.ts:62]: Done

File: src/modules/health/health.controller.ts (hunk 3 above)
Comment (karthikeyan-cere) [reply src/modules/health/health.controller.ts]: Done

**Review by evgeny-s — APPROVED** [review]

**Pull request merged**